### DISCUSSION TITLE
Position on Earth where the Moon is directly above

### DISCUSSION THREAD

**Discussion by mrtakaaxe** [cosinekitty/astronomy#373] [Q&A]
Can I use Position or Camera to work out where on earth, I guess longitude and latitude, that the moon is currently located (directly above)?  Essentially I'd like to know where/what you'd be seeing of earth, as an observer standing on the moon.  Is that possible?  Any assistance is much appreciated.  Hopefully this is not a dumb question.

Not sure if this helps but ChatGPT says that what I'm asking for is the Sublunar Point on Earth... Not sure if that's correct but it says that the steps for calculating this are:

1. Determine the moon’s right ascension and declination.
2. Calculate GST at the desired time.
3. Find LST using GST and the observer’s longitude.
4. Calculate the hour angle (HA) using LST and RA.
5. Convert HA to longitude

Latitude of the Sublunar Point = Moons declination
Longitude of the Subluner Point = - Hour Angle = Local Sidereal Time (equals Greenwich Sidereal Time + Observer Latitude) - Right Ascension

I'm totally confused where/how I use the Astronomy Engine to determine these elements.

**Comment by cosinekitty** [comment]
I think I can help here. I'm not sure which programming language you are using, but the names of the functions are similar across all of them. Take a look for your programming language documentation [here](https://github.com/cosinekitty/astronomy/tree/master?tab=readme-ov-file#supported-programming-languages) and click on "Code & Docs".

One quick caveat: the Moon's right ascension and declination is usually calculated for a particular observer on the surface of the Earth. This includes *topocentric* corrections that are significant for the Moon because of how close the Moon is to the Earth. That is how Astronomy Engine's "Equator" functions work.

The algorithm mentioned by ChatGPT sounds like it would need a geocentric position for the Moon, otherwise the longitude will not apply to some other location. Besides which, you shouldn't need to specify an observer's location; we are trying to find a location, not provide one! So Astronomy Engine's "Equator" won't work.

I'm thinking you should use one of the [GeoMoon](https://github.com/cosinekitty/astronomy/tree/master/source/csharp#astronomygeomoontime--astrovector) functions to find the vector from the center of the Earth to the center of the Moon. Then use [EquatorFromVector](https://github.com/cosinekitty/astronomy/tree/master/source/csharp#Astronomy.EquatorFromVector) to convert that vector to the geocentric declination and right ascension of the Moon.

Step 3 in the ChatGPT output is fishy to me. You should not need an observer's location, because we are trying to find a location from across the globe. Maybe it means that if you know any observer's longitude, and the hour angle of the Moon as seen from that observer's location, then you can subtract to find the longitude where the Moon is directly overhead? (In other words, where the Moon's hour angle is zero.)

Maybe this gets you a little closer?
